src/content/sidebar/sidebar-wrapper.tsx

```diff
@@ -1,59 +1,130 @@
+
     useEffect(() => {
         const messageCallback = (event) => {
             console.log('content-script container ev', event)
             if (event.type === 'update-panel-url') {
                 setUrl(event.url)
-            } else if (event.type === "toggle-sidebar") {
+            // unfortunatly the messages send by Window.postMessage() have a slightly 
+            // different structure than the messages send by the browser api
+            //
+            // we should probably make sure that the message was send by the
+            // url in `url` since every website can post a message like this
+            } else if (event.type === "toggle-sidebar" || event.data.type === "toggle-sidebar") {
                 setOpen(!isOpen)
             }
         }
 
+        if (isOpen) {
+            setShowIFrame(true)
+        } else {
+            ref.current.addEventListener('transitionend', () => {
+                if (ref.current.classList.contains('sidebar--closed')) {
+                    setShowIFrame(false)
+                    ref.current.removeListener('transitionend', this);
+                }
+            })
+            // hide iFrame after 5s in case transitions are disabled or not supported
+            setTimeout(() => {
+                if (ref.current.classList.contains('sidebar--closed')) {
+                    setShowIFrame(false)
+                }
+            }, 5000);
+        }
+
         browser.runtime.onMessage.addListener(messageCallback)
+        window.addEventListener('message', messageCallback)
 
-        return () => browser.runtime.onMessage.removeListener(messageCallback)
+        return () => {
+            browser.runtime.onMessage.removeListener(messageCallback)
+            window.removeEventListener('message', messageCallback)
+        }
     }, [isOpen, setOpen])
 
-    return <Slider
-        isOpen={isOpen}
-        width={450}
-        right
-        noOverlay
-        customBurgerIcon={false}
-        styles={styles}
-        // This is mainly here to ensure that when menu is closed
-        // by internal element logic we're aware and maintain proper state
-        onClose={() => setOpen(false)}
-    >
-        {/*{props.children}*/}
-        <iframe
-            css={css`
-                width: 100%;
-                height: 100%;
-            `}
-            src={url}/>
-    </Slider>
-}
-
-const styles = {
-    bmMenu: {
-        overflow: "hidden",
-    },
-    bmCross: {
-        background: '#bdc3c7',
-    },
-    bmMenuWrap: {
-        zIndex: 99999,
-    },
-}
+
+    const updateDrag = (event) => {
+        if (isDragging) {
+            const newWidth = event.clientX * 100 / window.innerWidth;
+            if (newWidth > 50) {
+                setLeft(false);
+                setWidth(100 - newWidth);
+            } else {
+                setLeft(true);
+                setWidth(newWidth);
+            }
+        }
+    }
+
+    const endDrag = () => {
+        if (isDragging) {
```

Comment:
nit: probably don't need the conditional, as it's ok to set it to false regardless

Reply:
right. thanks